comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector_test.go

```diff
@@ -60,7 +60,10 @@ func (s *mockServer) StreamEntities(_ *pbgo.ProcessStreamEntitiesRequest, out pb
 		}
 	}
 
-	return nil
+	// Block indefinitely to avoid closing the stream
+	for {
+		time.Sleep(1 * time.Minute)
+	}
```

Comment:
I'm wondering if we should introduce a `s.ctx, s.cancelFunc := context.WithCancel(context.Background())` in the mockServer. 

and instead of doing a `for loop`  we do a wait on `ctx.Done()` 

```suggestion
	<-s.ctx.Done()
```

we can add a `func (s *mockServer) Stop() { s.cancelFunc() } 
if in the test we want to properly stop the StreamEntities

Reply:
Good idea 👍 

Updated